File: Examples/CSharp/ModifyingAndConvertingImages/WebPImages/CreatingWebPImage.cs
using Aspose.Imaging.ImageOptions;
using Aspose.Imaging.Sources;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference when the project is built. Please see https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. If you prefer not to use NuGet, you can manually download Aspose.Imaging for .NET from https://releases.aspose.com/, install it, and then add its reference to this project. For any issues, questions, or suggestions, please feel free to contact us via https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.WebPImages
{
    class CreatingWebPImage
    {
        public static void Run()
        {
            Console.WriteLine("Running example CreatingWebPImage");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_WebPImages();

            // Create an instance of WebPOptions class and set properties
            WebPOptions imageOptions = new WebPOptions();
            imageOptions.Lossless = true;
            imageOptions.Source = new FileCreateSource(dataDir + "CreatingWebPImage_out.webp", false);

            // Create an instance of image class by using the WebPOptions instance that you have just created.
            using (Image image = Image.Create(imageOptions, 500, 500))
            {
                image.Save();
            }

            Console.WriteLine("Finished example CreatingWebPImage");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/WebPImages/CreatingWebPImage.vb

Imports Aspose.Imaging.FileFormats.Tiff.Enums
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.WebPImages
    Class CreatingWebPImage
        Public Shared Sub Run()
            ' ExStart:CreatingWebPImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_WebPImages()

            ' Create an instance of WebPOptions class
            Dim imageOptions As New WebPOptions()

            ' Set LossLess property
            imageOptions.Lossless = True

            ' Set Source property with the path and image file name where you want to create the WebP image.
            imageOptions.Source = New Sources.FileCreateSource(dataDir & Convert.ToString("CreatingWebPImage_out.webp"), False)

            ' Create an instance of image class by using WebOptions instance that you have just created.
            Using image__1 As Image = Image.Create(imageOptions, 500, 500)
                ' Save the image.
                image__1.Save()
            End Using
            ' ExEnd:CreatingWebPImage
        End Sub
    End Class
End Namespace